opengrok-tools/src/test/python/test_mirror.py

```diff
@@ -109,3 +114,59 @@ def test_valid_config():
     assert check_configuration({PROJECTS_PROPERTY:
                                 {"foo": {PROXY_PROPERTY: True}},
                                 PROXY_PROPERTY: "proxy"})
+
+
+@pytest.mark.skipif(not os.name.startswith("posix"), reason="requires posix")
+def test_incoming_retval(monkeypatch):
+    """
+    Test that the special CONTINUE_EXITVAL value bubbles all the way up to
+    the mirror.py return value.
+    """
+
+    class MockResponse:
+
+        # mock json() method always returns a specific testing dictionary
+        @staticmethod
+        def json():
+            return {"indexed": "true"}
+
+        @staticmethod
+        def raise_for_status():
+            pass
+
+    with tempfile.TemporaryDirectory() as source_root:
+        repo_name = "parent_repo"
+        repo_path = os.path.join(source_root, repo_name)
+        cloned_repo_name = "cloned_repo"
+        cloned_repo_path = os.path.join(source_root, cloned_repo_name)
+        project_name = "foo"    # does not matter for this test
+
+        os.mkdir(repo_path)
+
+        def mock_get_repos(*args, **kwargs):
+            from opengrok_tools.scm.repofactory import get_repository
+
+            return [get_repository(cloned_repo_path,
+                                   "git", project_name,
+                                   None, None, None, None)]
+
+        def mock_get(*args, **kwargs):
+            return MockResponse()
+
+        # Clone a Git repository so that it can pull.
+        import pygit2
+        pygit2.init_repository(repo_path, True)
+        pygit2.clone_repository(repo_path, cloned_repo_path)
```

Comment:
I'm not that convinced that it is worth bringing this dependency for simple `git init` and `git clone` commands call.

Reply:
I think both approaches have their pros and cons. I used pygit2 mainly because it is an API (and APIs are out there for consumption), and widely used one. If I were to use command execution, I'd have to deal with return codes, error strings and rely on the command line interface to stay stable which is arguably less likely than the programmatic API. Also, this should encourage anyone to add more tests to the mirroring - which is minimally covered - seeing how easy it is.

Reply:
fair enough